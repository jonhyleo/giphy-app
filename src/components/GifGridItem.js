import PropTypes from "prop-types";

const GifGridItem = ({ id, title, url }) => {
  return (
    <>
      <div className="col-sm-6 col-lg-4" key={id}>
        <div className="card mb-4">
          <div className="card-img-wrapper">
            <img src={url} alt={title} className="card-img-top" />
          </div>
          <div className="card-body">
            <p className="card-text">{title}</p>
          </div>
        </div>
      </div>
    </>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
