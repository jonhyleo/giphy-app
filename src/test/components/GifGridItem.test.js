import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Test for GifGridItem component", () => {
  const title = "Title";
  const url = "https://localhost/image.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("The component must be render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Component must have a paragraph", () => {
    const p = wrapper.find("p");

    expect(p.text().trim()).toBe(title);
  });

  test("Image src and alt must match values provided on props", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
});
