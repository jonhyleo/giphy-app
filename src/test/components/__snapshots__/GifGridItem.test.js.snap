// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for GifGridItem component The component must be render properly 1`] = `
<Fragment>
  <div
    className="col-sm-6 col-lg-4"
  >
    <div
      className="card mb-4"
    >
      <div
        className="card-img-wrapper"
      >
        <img
          alt="Title"
          className="card-img-top"
          src="https://localhost/image.jpg"
        />
      </div>
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          Title
        </p>
      </div>
    </div>
  </div>
</Fragment>
`;
